File: Application/DTOs/Order/CreateOrderRequest.cs
using Common.Constants;
using System.ComponentModel.DataAnnotations;

namespace Application.DTOs.Order;

public class CreateOrderRequest
{

    [Required]
    [MinLength(1)]
    public string DeliveryAddress { get; set; } = null!;

    [Range(-90, 90)]
    public double? DeliveryLatitude { get; set; }

    [Range(-180, 180)]
    public double? DeliveryLongitude { get; set; }

    public bool PayOnline { get; set; }

    [Required]
    [MinLength(1)]
    [MaxLength(OrderValidationConstants.MaxOrderLineCount)]
    public List<CreateOrderItemDto> Items { get; set; } = new();
}

File: WebApp/src/app/features/orders/models/create-order-request.ts
export const MAX_ORDER_LINE_COUNT = 50;
export const MAX_ORDER_TOTAL_QUANTITY = 200;

export interface CreateOrderItemRequest {
  readonly menuItemId: string;
  readonly quantity: number;
}

export interface CreateOrderRequest {
  readonly deliveryAddress: string;
  readonly deliveryLatitude: number | null;
  readonly deliveryLongitude: number | null;
  readonly payOnline: boolean;
  readonly items: CreateOrderItemRequest[];
}
